TestDateTimeUtils fails if system timezone has DST in the future

```
/usr/java/jdk1.8/bin/java -ea -Xmx128m -Duser.timezone=America/New_York ... org.h2.test.unit.TestDateTimeUtils
17:29:55 529:14:43.601 org.h2.test.unit.TestDateTimeUtils Expected: 7200000000000 actual: 10800000000000
Exception in thread "main" java.lang.AssertionError: Expected: 7200000000000 actual: 10800000000000
	at org.h2.test.TestBase.fail(TestBase.java:328)
	at org.h2.test.TestBase.assertEquals(TestBase.java:764)
	at org.h2.test.unit.TestDateTimeUtils.testUTC2ValueImpl(TestDateTimeUtils.java:177)
	at org.h2.test.unit.TestDateTimeUtils.testUTC2Value(TestDateTimeUtils.java:153)
	at org.h2.test.unit.TestDateTimeUtils.test(TestDateTimeUtils.java:64)
	at org.h2.test.unit.TestDateTimeUtils.main(TestDateTimeUtils.java:55)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.intellij.rt.execution.application.AppMainV2.main(AppMainV2.java:131)

--- a/h2/src/docsrc/html/changelog.html
+++ b/h2/src/docsrc/html/changelog.html
@@ -21,6 +21,8 @@ Change Log
 
 <h2>Next Version (unreleased)</h2>
 <ul>
+<li>#2335: TestDateTimeUtils fails if system timezone has DST in the future
+</li>
 <li>Issue #2330: Syntax error with parenthesized expression in GROUP BY clause
 </li>
 <li>Issue #2256: &lt;interval value expression&gt; with datetime subtraction

--- a/h2/src/test/org/h2/test/unit/TestDateTimeUtils.java
+++ b/h2/src/test/org/h2/test/unit/TestDateTimeUtils.java
@@ -141,7 +141,10 @@ public class TestDateTimeUtils extends TestBase {
     private void testUTC2Value(boolean allTimeZones) {
         TimeZone def = TimeZone.getDefault();
         GregorianCalendar gc = new GregorianCalendar();
-        String[] ids = allTimeZones ? TimeZone.getAvailableIDs() : new String[] { def.getID(), "+10" };
+        String[] ids = allTimeZones ? TimeZone.getAvailableIDs()
+                : new String[] { def.getID(), "+10",
+                        // Any time zone with DST in the future (JDK-8073446)
+                        "America/New_York" };
         try {
             for (String id : ids) {
                 if (allTimeZones) {
@@ -171,7 +174,7 @@ public class TestDateTimeUtils extends TestBase {
             for (int j = 0; j < 48; j++) {
                 gc.set(year, month - 1, day, j / 2, (j & 1) * 30, 0);
                 long timeMillis = gc.getTimeInMillis();
-                ValueTimestamp ts = ValueTimestamp.get(gc.getTimeZone(), new Timestamp(timeMillis));
+                ValueTimestamp ts = ValueTimestamp.get(null, new Timestamp(timeMillis));
                 timeMillis += DateTimeUtils.getTimeZoneOffsetMillis(timeMillis);
                 assertEquals(ts.getDateValue(), DateTimeUtils.dateValueFromLocalMillis(timeMillis));
                 assertEquals(ts.getTimeNanos(), DateTimeUtils.nanosFromLocalMillis(timeMillis));

